lightning/src/routing/scorer.rs

```diff
@@ -85,12 +107,48 @@ pub struct ScoringParameters {
 	/// [`failure_penalty_half_life`]: Self::failure_penalty_half_life
 	pub failure_penalty_msat: u64,
 
-	/// The time needed before any accumulated channel failure penalties are cut in half.
-	#[cfg(not(feature = "no-std"))]
+	/// The time required to elapse before any accumulated channel failure penalties are cut in
+	/// half.
+	///
+	/// # Note
+	///
+	/// When time is an [`Eternity`], as is default when enabling feature `no-std`, it will never
+	/// elapse. Therefore, this penalty will never decay.
```

Comment:
That wouldn't be accurate since `failure_penalty_msat` is added each time the channel fails to relay a payment, i.e., "this penalty" is referring to "any accumulated channel failure penalties". I've re-worded a bit and included a link to `failure_penalty_msat`.